src/csharp/Grpc.Core.Tests/CallCancellationTest.cs

```diff
@@ -189,5 +189,41 @@ public void CanDisposeDefaultCancellationRegistration()
 
             using (obj) {}
         }
+
+        [Test]
+        public async Task ServerStreamingCall_CancelAfterServerHasFinished()
```

Comment:
I suggest running the test many times (e.g. 5000) to make sure it's not flaky
(you can use run_tests.py -n 5000).

Reply:
Ran 5000 times OK